dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/sms/TrackedEntityRegistrationSMSListener.java

```diff
@@ -164,4 +179,60 @@ private TrackedEntityAttributeValue createTrackedEntityAttributeValue(
     trackedEntityAttributeValue.setValue(value);
     return trackedEntityAttributeValue;
   }
+
+  private void enrollTrackedEntity(
+      TrackedEntity trackedEntity,
+      Program program,
+      OrganisationUnit organisationUnit,
+      Date occurredDate) {
+    Enrollment enrollment =
+        prepareEnrollment(
+            trackedEntity,
+            program,
+            new Date(),
+            occurredDate,
+            organisationUnit,
+            CodeGenerator.generateUid());
+    manager.save(enrollment);
+    trackerOwnershipAccessManager.assignOwnership(
+        trackedEntity, program, organisationUnit, true, true);
+    eventPublisher.publishEvent(new ProgramEnrollmentNotificationEvent(this, enrollment.getId()));
+    manager.update(enrollment);
+    trackedEntityService.updateTrackedEntity(trackedEntity);
+  }
+
+  private Enrollment prepareEnrollment(
```

Comment:
Please inline this method is used only once and its name does not add value.

It even makes things more complex as

```java
    enrollment.setUid(CodeGenerator.isValidUid(uid) ? uid : CodeGenerator.generateUid());
```

has dead code as the `uid` is always `CodeGenerator.generateUid()`. Same with `enrollmentDate` which is set to `new Date()`.

Reply:
I'll take a look at the redundant parameters though!